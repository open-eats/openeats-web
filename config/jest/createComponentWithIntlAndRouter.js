import React from 'react';
import renderer from 'react-test-renderer';
import { IntlProvider } from 'react-intl';
import { MemoryRouter } from 'react-router-dom'

const createComponentWithIntl = (children, props = { locale: 'en' }) => {
  return renderer.create(
    <IntlProvider {...props}>
      <MemoryRouter>
        { children }
      </MemoryRouter>
    </IntlProvider>
  );
};

export default createComponentWithIntl;