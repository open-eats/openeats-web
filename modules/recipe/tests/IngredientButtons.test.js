import React from 'react';
import RecipeListStatusConstants from '../constants/RecipeListStatusConstants';
import IngredientButtons from '../components/IngredientButtons';
import createComponentWithIntlAndRouter from '../../../config/jest/createComponentWithIntlAndRouter';

const lists = [{id: 1, title: 'title'},{id: 2, title: 'tÃ¼tle'},];

test('Ingredient Buttons Default', () => {
  const bulkAdd = jest.fn();
  const checkAll = jest.fn();
  const unCheckAll = jest.fn();
  const component = createComponentWithIntlAndRouter(
    <IngredientButtons
      listStatus={ '' }
      lists={ lists }
      bulkAdd={ bulkAdd }
      checkAll={ checkAll }
      unCheckAll={ unCheckAll }
    />
  );
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

test('Ingredient Buttons Loading', () => {
  const bulkAdd = jest.fn();
  const checkAll = jest.fn();
  const unCheckAll = jest.fn();
  const component = createComponentWithIntlAndRouter(
    <IngredientButtons
      listStatus={ RecipeListStatusConstants.LOADING }
      lists={ lists }
      bulkAdd={ bulkAdd }
      checkAll={ checkAll }
      unCheckAll={ unCheckAll }
    />
  );
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

test('Ingredient Buttons Complete', () => {
  const bulkAdd = jest.fn();
  const checkAll = jest.fn();
  const unCheckAll = jest.fn();
  const component = createComponentWithIntlAndRouter(
    <IngredientButtons
      listStatus={ RecipeListStatusConstants.COMPLETE }
      lists={ lists }
      bulkAdd={ bulkAdd }
      checkAll={ checkAll }
      unCheckAll={ unCheckAll }
    />
  );
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

test('Ingredient Buttons Error', () => {
  const bulkAdd = jest.fn();
  const checkAll = jest.fn();
  const unCheckAll = jest.fn();
  const component = createComponentWithIntlAndRouter(
    <IngredientButtons
      listStatus={ RecipeListStatusConstants.ERROR }
      lists={ lists }
      bulkAdd={ bulkAdd }
      checkAll={ checkAll }
      unCheckAll={ unCheckAll }
    />
  );
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
