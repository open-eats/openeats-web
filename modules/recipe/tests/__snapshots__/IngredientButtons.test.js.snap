// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Ingredient Buttons Complete 1`] = `
<div
  className="ingredients-save"
>
  <div
    className="btn-group"
    role="group"
  >
    <button
      className="btn btn-default"
      onClick={[MockFunction]}
    >
      Check All
    </button>
    <button
      className="btn btn-default"
      onClick={[MockFunction]}
    >
      Clear
    </button>
    <div
      className="dropdown btn-group"
    >
      <button
        aria-expanded={false}
        aria-haspopup={true}
        className="dropdown-toggle btn btn-default"
        disabled={false}
        id="ing-save"
        onClick={[Function]}
        onKeyDown={[Function]}
        role="button"
        type="button"
      >
        Add To
         
        <span
          className="caret"
        />
      </button>
      <ul
        aria-labelledby="ing-save"
        className="dropdown-menu"
        role="menu"
      >
        <li
          className=""
          role="presentation"
        >
          <a
            href="#"
            onClick={[Function]}
            onKeyDown={[Function]}
            role="menuitem"
            tabIndex="-1"
          >
            title
          </a>
        </li>
        <li
          className=""
          role="presentation"
        >
          <a
            href="#"
            onClick={[Function]}
            onKeyDown={[Function]}
            role="menuitem"
            tabIndex="-1"
          >
            t端tle
          </a>
        </li>
      </ul>
    </div>
  </div>
  <div
    className="glyphicon glyphicon-ok"
  />
</div>
`;

exports[`Ingredient Buttons Default 1`] = `
<div
  className="ingredients-save"
>
  <div
    className="btn-group"
    role="group"
  >
    <button
      className="btn btn-default"
      onClick={[MockFunction]}
    >
      Check All
    </button>
    <button
      className="btn btn-default"
      onClick={[MockFunction]}
    >
      Clear
    </button>
    <div
      className="dropdown btn-group"
    >
      <button
        aria-expanded={false}
        aria-haspopup={true}
        className="dropdown-toggle btn btn-default"
        disabled={false}
        id="ing-save"
        onClick={[Function]}
        onKeyDown={[Function]}
        role="button"
        type="button"
      >
        Add To
         
        <span
          className="caret"
        />
      </button>
      <ul
        aria-labelledby="ing-save"
        className="dropdown-menu"
        role="menu"
      >
        <li
          className=""
          role="presentation"
        >
          <a
            href="#"
            onClick={[Function]}
            onKeyDown={[Function]}
            role="menuitem"
            tabIndex="-1"
          >
            title
          </a>
        </li>
        <li
          className=""
          role="presentation"
        >
          <a
            href="#"
            onClick={[Function]}
            onKeyDown={[Function]}
            role="menuitem"
            tabIndex="-1"
          >
            t端tle
          </a>
        </li>
      </ul>
    </div>
  </div>
  
</div>
`;

exports[`Ingredient Buttons Error 1`] = `
<div
  className="ingredients-save"
>
  <div
    className="btn-group"
    role="group"
  >
    <button
      className="btn btn-default"
      onClick={[MockFunction]}
    >
      Check All
    </button>
    <button
      className="btn btn-default"
      onClick={[MockFunction]}
    >
      Clear
    </button>
    <div
      className="dropdown btn-group"
    >
      <button
        aria-expanded={false}
        aria-haspopup={true}
        className="dropdown-toggle btn btn-default"
        disabled={false}
        id="ing-save"
        onClick={[Function]}
        onKeyDown={[Function]}
        role="button"
        type="button"
      >
        Add To
         
        <span
          className="caret"
        />
      </button>
      <ul
        aria-labelledby="ing-save"
        className="dropdown-menu"
        role="menu"
      >
        <li
          className=""
          role="presentation"
        >
          <a
            href="#"
            onClick={[Function]}
            onKeyDown={[Function]}
            role="menuitem"
            tabIndex="-1"
          >
            title
          </a>
        </li>
        <li
          className=""
          role="presentation"
        >
          <a
            href="#"
            onClick={[Function]}
            onKeyDown={[Function]}
            role="menuitem"
            tabIndex="-1"
          >
            t端tle
          </a>
        </li>
      </ul>
    </div>
  </div>
  <div
    className="glyphicon glyphicon-remove"
  />
</div>
`;

exports[`Ingredient Buttons Loading 1`] = `
<div
  className="ingredients-save"
>
  <div
    className="btn-group"
    role="group"
  >
    <button
      className="btn btn-default"
      onClick={[MockFunction]}
    >
      Check All
    </button>
    <button
      className="btn btn-default"
      onClick={[MockFunction]}
    >
      Clear
    </button>
    <div
      className="dropdown btn-group"
    >
      <button
        aria-expanded={false}
        aria-haspopup={true}
        className="dropdown-toggle btn btn-default"
        disabled={false}
        id="ing-save"
        onClick={[Function]}
        onKeyDown={[Function]}
        role="button"
        type="button"
      >
        Add To
         
        <span
          className="caret"
        />
      </button>
      <ul
        aria-labelledby="ing-save"
        className="dropdown-menu"
        role="menu"
      >
        <li
          className=""
          role="presentation"
        >
          <a
            href="#"
            onClick={[Function]}
            onKeyDown={[Function]}
            role="menuitem"
            tabIndex="-1"
          >
            title
          </a>
        </li>
        <li
          className=""
          role="presentation"
        >
          <a
            href="#"
            onClick={[Function]}
            onKeyDown={[Function]}
            role="menuitem"
            tabIndex="-1"
          >
            t端tle
          </a>
        </li>
      </ul>
    </div>
  </div>
  <div
    className="sk-circle-wrapper sk-spinner recipe-list-spinner"
  >
    <div
      className="sk-circle1 sk-circle"
    />
    <div
      className="sk-circle2 sk-circle"
    />
    <div
      className="sk-circle3 sk-circle"
    />
    <div
      className="sk-circle4 sk-circle"
    />
    <div
      className="sk-circle5 sk-circle"
    />
    <div
      className="sk-circle6 sk-circle"
    />
    <div
      className="sk-circle7 sk-circle"
    />
    <div
      className="sk-circle8 sk-circle"
    />
    <div
      className="sk-circle9 sk-circle"
    />
    <div
      className="sk-circle10 sk-circle"
    />
    <div
      className="sk-circle11 sk-circle"
    />
    <div
      className="sk-circle12 sk-circle"
    />
  </div>
</div>
`;
